rispy/parser.py

```diff
@@ -1,67 +1,126 @@
+                                            `is_header` method. Defaults to
+                                            `False`.
+        SKIP_UNKNOWN_TAGS (bool, optional): Bool to skip tags that are not in
+                                            `TAG_KEY_MAPPING`. If unknown tags
+                                            are not skipped, they will be added
+                                            to the `unknown_tag` key.
+                                            Defaults to `False`.
+        ENFORCE_LIST_TAGS (bool, optional): Bool for choosing whether to
+                                            strictly enforce list type tags.
+                                            If this is `False`, tags that
+                                            occur mutliple times in a reference
+                                            will be converted to a list instead
+                                            of being overriden. Values set to
+                                            be list tags will still be read as
+                                            list tags. Defaults to `True`.
+        DEFAULT_MAPPING (dict): A default mapping for the custom parser.
+                                Required.
+        DEFAULT_LIST_TAGS (list): A list of tags that should be read as lists.
+                                  Required.
+
+    Class methods:
+        get_content: Returns the non-tag part of a line. Required.
+        is_header: Returns a bool for whether a line is a header and should be
+                   skipped. This method only operates on lines outside of a
+                   reference. Defaults to `False` for all lines.
+        get_tag: Returns the tag part of a line. Default is to return the
+                 first two characters.
+        is_tag: Determines whether a line has a tag, returning a bool. Uses
+                regex in `PATTERN` by default.
+        clean_text: Clean the text body before parsing begins. By default,
+                    it removes UTF-BOM characters.
+
+    """
+
+    START_TAG: str
     END_TAG: str = "ER"
     IGNORE: List[str] = []
-    PATTERN: str = None
+    PATTERN: str
+    SKIP_MISSING_TAGS: bool = False
+    SKIP_UNKNOWN_TAGS: bool = False
+    ENFORCE_LIST_TAGS: bool = True
+    DEFAULT_MAPPING: Dict
+    DEFAULT_LIST_TAGS: List[str]
 
-    def __init__(self, lines, mapping, strict=True):
-        self.lines = lines
+    def __init__(self, *, mapping: Optional[Dict] = None, list_tags: Optional[List] = None):
+        """Initialize the parser function.
+
+        Args:
+            mapping (dict, optional): Map tags to tag names.
+            list_tags (list, optional): List of list-type tags.
+
+        """
         self.pattern = re.compile(self.PATTERN)
-        self._mapping = mapping
-        self.strict = strict
+        self.mapping = mapping or self.DEFAULT_MAPPING
+        self.list_tags = list_tags or self.DEFAULT_LIST_TAGS
```

Comment:
@J535D165 was correct here; if list_tags is falsy, like [], then this will use the default, even if you intentionally specified no skip tags - perhaps change to 

```python
self.mapping = mapping if mapping is not None else self. DEFAULT_MAPPING
self.list_tags = list_tags if list_tags is not None else self. DEFAULT_LIST_TAGS
```